### PR TITLE
nvuApp: login page

### PR DESCRIPTION
**Pull request by denisdimitrov**
Login page with full functionality and tests. Also added one more validation for the TutorHandler.

### CODE REVIEW

File: src/main/java/http/controllers/LoginHandler.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  public LoginHandler(TutorRepository repository) {
+    this.repository = repository;
+  }
+
+  @Override
+  public Response handle(Request req) {
+    String tutorID = req.param("LoginID");
+    String password = req.param("LoginPass");
+    List<Tutor> tutor = repository.findTutor(tutorID);
+    if (Strings.isNullOrEmpty(tutorID) && Strings.isNullOrEmpty(password)) {
```
Comment (mgenov) [inline src/main/java/http/controllers/LoginHandler.java]: These checks need to be simplified somehow. Code looks really messy and not understandable.

File: src/test/java/com/clouway/nvuapp/adapter/PersistentQuestionRepositoryTest.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * @author Vasil Mitov <v.mitov.clouway@gmail.com>
  */
 public class PersistentQuestionRepositoryTest {
-  private DataStore datastore = new DataStore(new ConnectionProvider("nvuApp", "clouway.com", "localhost"));
+  private DataStore datastore = new DataStore(new ConnectionProvider("nvuApp", "root", "localhost"));
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/adapter/PersistentQuestionRepositoryTest.java]: This is going to clean up any tests data (used during development) when tests are run. I suppose separate databases need to be used for local testing and for tests.

File: src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }});
+    Response response = loginHandler.handle(request);
+    assertThat(reader().read(response), containsString("<h1>Добре дошли.</h1>"));
+  }
+
+  @Test
+  public void invalidIdAndPass() throws Exception {
+    tutors.add(new Tutor("admin", "admin"));
+    context.checking(new Expectations() {{
+      oneOf(repository).findTutor("guest");
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java]: It's strange findTutor to be executed event before all params are taken from the request.

File: src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>.
+ */
+public class LoginHandlerTest {
+  private JUnitRuleMockery context = new JUnitRuleMockery();
+  private TutorRepository repository = context.mock(TutorRepository.class);
+  private LoginHandler loginHandler = new LoginHandler(repository);
+  private Request request = context.mock(Request.class);
```
Comment (mgenov) [inline src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java]: Using of FakeRequest will simplify tests, cause you don't have to stub the param calls. 

The following snippet is much shorter and easier to read...
```java
new FakeRequest(ImmutableMap.of("LoginID", "admin", "LoginPass", "admin"));
```
then the used one: 

```java
  context.checking(new Expectations() {{
      oneOf(repository).findTutor("admin");
      will(returnValue(tutors));
      oneOf(request).param("LoginID");
      will(returnValue("admin"));
      oneOf(request).param("LoginPass");
      will(returnValue("admin"));
    }});
```

File: src/test/java/persistent/PersistentTutorRepositoryTest.java [hunk 5]
```diff
@@ -20,7 +20,7 @@
  * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>.
  */
 public class PersistentTutorRepositoryTest {
-  private ConnectionProvider connectionProvider = new ConnectionProvider("nvuApp","clouway.com","localhost");
+  private ConnectionProvider connectionProvider = new ConnectionProvider("nvuApp","root","localhost");
```
Comment (mgenov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: This exposure is going to duplicate connection string in multiple tests which is a smell. 

Check #17 for details.

File: src/main/java/http/controllers/LoginHandler.java (hunk 1 above)
Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: I will try to make it more readable and simple.

File: src/test/java/com/clouway/nvuapp/adapter/PersistentQuestionRepositoryTest.java (hunk 2 above)
Comment (denisdimitrov) [inline src/test/java/com/clouway/nvuapp/adapter/PersistentQuestionRepositoryTest.java]: I see, I will make another db for the tests.

File: src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java (hunk 4 above)
Comment (denisdimitrov) [inline src/test/java/com/clouway/nvuapp/core/LoginHandlerTest.java]: I will simplify the tests, both these and the tutor.

File: src/test/java/persistent/PersistentTutorRepositoryTest.java (hunk 5 above)
Comment (mgenov) [inline src/test/java/persistent/PersistentTutorRepositoryTest.java]: @denisdimitrov @martinmilev and @gadjevb could give you some light on this topic.

File: src/main/java/http/controllers/LoginHandler.java (hunk 1 above)
Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: Does this look better? Or should I think of something else.

screenshot from 2017-01-12 18-27-39

Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: Just noticed my mistake, isEmpty validation shouldn't have !.

Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: Better?
screenshot from 2017-01-12 18-27-39

Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: I hope this is better.
screenshot from 2017-01-12 18-31-19

Comment (denisdimitrov) [inline src/main/java/http/controllers/LoginHandler.java]: I know that it's my own implementation of the guava isNullOrEmpty() method but it probably looks better. Waiting for feedback.
screenshot from 2017-01-12 18-36-29

**Comment by denisdimitrov** [comment]
I closed it because I had some major problems with git. I intent to reopen it with all of your change requests.

**Pull request closed without merging**